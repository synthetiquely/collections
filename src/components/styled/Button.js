import React from 'react';
import glamorous from 'glamorous';
import Spinner from './Spinner';

const Button = glamorous.button(
  {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    fontSize: '16px',
    padding: '10px',
    color: '#000',
    border: '1px solid #ccc',
    outline: 'none',
    backgroundColor: '#fff',
    cursor: 'pointer',
    ':hover,:active,:focus': {
      borderLeft: '1px solid',
      borderColor: '#ffdb4d',
    },
  },
  ({ size, inverted, bordered }) => {
    let styles = {};
    if (size === 'fullWidth') {
      styles = {
        ...styles,
        width: '100%',
      };
      if (inverted) {
        styles = {
          ...styles,
          ':hover,:active,:focus': {
            backgroundColor: '#ffdb4d',
            color: '#fff',
            borderColor: '#ffd35f',
          },
        };
      }
      if (bordered) {
        styles = {
          ...styles,
          borderRadius: '10px',
        };
      }
    }
    return styles;
  },
);

export default ({ loading, ...rest }) => (
  <Button {...rest}>
    {loading && <Spinner size="sm" />}{' '}
    <span style={{ padding: '0 15px' }}>{rest.children}</span>
  </Button>
);
