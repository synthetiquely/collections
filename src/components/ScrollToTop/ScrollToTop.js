import React, { Component } from 'react';
import { css } from 'glamor';
import Button from '../styled/Button';
import Icon from '../styled/Icon';
import { SCROLL_TO_TOP_LIMIT } from '../../constants';

const styles = css({
  position: 'fixed',
  bottom: '20px',
  right: '30px',
  zIndex: '10',
});

class ScrollToTop extends Component {
  constructor(props) {
    super(props);
    this.state = {
      show: false,
    };
    this.onClick = this.onClick.bind(this);
    this.onScroll = this.onScroll.bind(this);
  }

  componentDidMount() {
    document.addEventListener('scroll', this.onScroll, {
      passive: true,
      capture: true,
    });
  }

  componentWillUnmount() {
    document.removeEventListener('scroll', this.onScroll, true);
  }

  onScroll() {
    if (
      document.body.scrollTop > SCROLL_TO_TOP_LIMIT ||
      document.documentElement.scrollTop > SCROLL_TO_TOP_LIMIT
    ) {
      if (!this.state.show) {
        this.setState({
          show: true,
        });
      }
    } else if (this.state.show) {
      this.setState({
        show: false,
      });
    }
  }

  onClick = () => {
    window.scroll({
      top: 0,
      behavior: 'smooth',
    });
  };

  render() {
    const { show } = this.state;
    if (show && this.props.show) {
      return (
        <div className={styles} title="Вернуться наверх">
          <Button bordered inverted type="button" onClick={this.onClick}>
            <Icon size="md">&uarr;</Icon>
          </Button>
        </div>
      );
    }

    return null;
  }
}

export default ScrollToTop;
