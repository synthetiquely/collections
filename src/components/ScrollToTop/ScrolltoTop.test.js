import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import ScrollToTop from './ScrollToTop';
import Button from '../styled/Button';

describe('ScrollToTop Component', () => {
  it('should render and match its own snapshot', () => {
    const wrapper = mount(<ScrollToTop />);
    const tree = renderer.create(<ScrollToTop />).toJSON();

    expect(wrapper).toHaveLength(1);
    expect(tree).toMatchSnapshot();
  });

  it('should render nothing, if show props is false', () => {
    const wrapper = mount(<ScrollToTop />);

    expect(wrapper.find(Button)).toHaveLength(0);
  });

  it('should render a button, if show props is true', () => {
    const wrapper = mount(<ScrollToTop show />);
    wrapper.setState({ show: true });

    expect(wrapper.find(Button)).toHaveLength(1);
  });
});
