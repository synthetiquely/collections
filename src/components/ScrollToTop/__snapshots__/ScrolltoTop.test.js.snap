// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTop Component should render and match its own snapshot 1`] = `null`;
